backend/workflow_manager/endpoint/destination.py

```diff
@@ -140,37 +163,34 @@ def handle_output(
             if connection_type == WorkflowEndpoint.ConnectionType.API:
                 self._handle_api_result(file_name=file_name, error=error, result=result)
             return
+        file_history = FileHistoryHelper.get_file_history(
+            workflow=workflow, cache_key=file_hash.file_hash
+        )
         if connection_type == WorkflowEndpoint.ConnectionType.FILESYSTEM:
             self.copy_output_to_output_directory()
         elif connection_type == WorkflowEndpoint.ConnectionType.DATABASE:
-            self.insert_into_db(
-                file_history=file_history, input_file_path=input_file_path
-            )
+            if (
```

Comment:
@vishnuszipstack  can we  remove this nesting.

Reply:
done